.github/workflows/docs.yml

```diff
@@ -45,6 +51,20 @@ jobs:
           path: build_docs/html
           name: docs-develop
 
+  deploy:
+    needs: build
+    if: github.event_name != 'pull_request' && github.ref == 'refs/heads/develop'
```

Comment:
Unfortunately, GH pages don't support `preview` in PRs (which would be amazing).
But it's still better to build docs in PRs before merging, rather than have an issue later

Reply:
I think if doxygen pre-commit check passed than this will work fine. Github pages is a documentation for developers, it is not a big issue if it is outdated for some time while. 

The number of CI runners is limited, so I would prefer to not spend them on this. Also I don't remember that we ever had this workflow failed. https://github.com/XRPLF/clio/actions/workflows/docs.yml?query=is%3Afailure

Reply:
1. I'm using `ubuntu-latest` to build the image, so it doesn't affect the number of CI runners.
2. We didn't have the workflow fail because we were ignoring all the warnings.

Reply:
The number of `ubuntu-latest` running simultaneously is still limited.